File: src/main/java/me/kuku/yuq/entity/NeTeaseEntity.java
package me.kuku.yuq.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Table(name = "neTease")
@AllArgsConstructor
@NoArgsConstructor
@Data
public class NeTeaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column(unique = true)
    private Long qq;
    private String MUSIC_U;
    private String __csrf;

    public NeTeaseEntity(String MUSIC_U, String __csrf){
        this.MUSIC_U = MUSIC_U;
        this.__csrf = __csrf;
    }

    public String getCookie(){
        return String.format("os=pc; osver=Microsoft-Windows-10-Professional-build-10586-64bit; appver=2.0.3.131777; channel=netease; __remember_me=true; MUSIC_U=%s; __csrf=%s; ", MUSIC_U, __csrf);
    }

    public NeTeaseEntity(Long qq){
        this.qq = qq;
    }
}

File: src/main/kotlin/me/kuku/yuq/entity/NetEaseEntity.kt
package me.kuku.yuq.entity

import org.hibernate.annotations.Type
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Service
import javax.persistence.*

@Entity
@Table(name = "net_ease")
class NetEaseEntity: BaseEntity() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int? = null
    @OneToOne
    @JoinColumn(name = "qq_id")
    var qqEntity: QqEntity? = null
    @Column(length = 2000)
    var musicU: String = ""
    @Column(length = 2000)
    var csrf: String = ""
    @Type(type = "json")
    @Column(columnDefinition = "json")
    var config: NetEaseConfig = NetEaseConfig()

    fun cookie() = "os=pc; osver=Microsoft-Windows-10-Professional-build-10586-64bit; appver=2.0.3.131777; channel=netease; __remember_me=true; MUSIC_U=$musicU; __csrf=$csrf; "
}

interface NetEaseRepository: JpaRepository<NetEaseEntity, Int> {
    fun findByQqEntity(qqEntity: QqEntity): NetEaseEntity?
}

@Service
class NetEaseService (
    private val netEaseRepository: NetEaseRepository
) {
    fun save(netEaseEntity: NetEaseEntity): NetEaseEntity = netEaseRepository.save(netEaseEntity)

    fun findByQqEntity(qqEntity: QqEntity) = netEaseRepository.findByQqEntity(qqEntity)

    fun findAll(): List<NetEaseEntity> = netEaseRepository.findAll()
}

data class NetEaseConfig(var sign: Status = Status.OFF, var musicianSign: Status = Status.OFF)